### ISSUE TITLE
Gateway-proxy is crashing during startup

### ISSUE THREAD

**Issue by dakamarko** [solo-io/gloo#5980]
### Gloo Edge Version

1.10.x (latest stable)

### Kubernetes Version

1.21.x

### Describe the bug

After upgrading to Gloo Edge 1.10.10 gateway-proxy can NOT get up. 
Earlier version like 1.10.8 works fine.

I've noticed three situation in stdout:
1. No output at all
2. tcmalloc like:
```external/com_github_google_tcmalloc/tcmalloc/internal/percpu_tcmalloc.h:729] !hdr.IsLocked() @ 0x555558aceefb 0x555558aa9b28 0x555558aa9 │
 │ 758 0x555558aa85dd 0x555558aa8166 0x555558aa59ca 0x555558b1b8aa 0x555557c4ff18 0x55555825431d 0x55555837f108 0x55555837ddf1 0x555557c6ed │
 │ 9d 0x5555585aff63 0x7ffff7d5b49a                                                                                                                                                                                                    │
 │ [300][critical][backtrace] [external/envoy/source/server/backtrace.h:92] Envoy version: 78625292510e31f0affd531 │
 │ 1a2957c5ab2abd55c/1.20.1/Distribution/RELEASE/BoringSSL                                                                                  │
 │300][critical][backtrace] [external/envoy/source/server/backtrace.h:98] #0: [0x7ffff7d10320]```
 
 3.
 ```
 ....all init logs that looks fine..then...
 
 [2022-02-25 07:49:42.301][7][info][main] [external/envoy/source/server/server.cc:817] all clusters initialized. initializing init manager
[2022-02-25 07:49:42.332][7][info][upstream] [external/envoy/source/server/lds_api.cc:77] lds: add/update listener 'listener-::-443'
[2022-02-25 07:49:42.333][7][info][upstream] [external/envoy/source/server/lds_api.cc:77] lds: add/update listener 'listener-::-5080'
[2022-02-25 07:49:42.344][7][info][upstream] [external/envoy/source/server/lds_api.cc:77] lds: add/update listener 'listener-::-5443'
[2022-02-25 07:49:42.345][7][info][upstream] [external/envoy/source/server/lds_api.cc:77] lds: add/update listener 'listener-::-80'
[2022-02-25 07:49:42.374][7][info][config] [external/envoy/source/server/listener_manager_impl.cc:779] all dependencies initialized. starting workers
[2022-02-25 07:49:42.382][89][critical][backtrace] [external/envoy/source/server/backtrace.h:104] Caught Segmentation fault, suspect faulting address 0x0
[2022-02-25 07:49:42.382][89][critical][backtrace] [external/envoy/source/server/backtrace.h:91] Backtrace (use tools/stack_decode.py to get line numbers):
[2022-02-25 07:49:42.382][89][critical][backtrace] [external/envoy/source/server/backtrace.h:92] Envoy version: 78625292510e31f0affd5311a2957c5ab2abd55c/1.20.1/Distribution>
external/com_github_google_tcmalloc/tcmalloc/internal/percpu_tcmalloc.h:729] !hdr.IsLocked() @ 0x555558aceefb 0x555558aa9b28 0x555558aa9758 0x555558aa85dd 0x555558aa8ba7 0x>
[2022-02-25 07:49:42.382][62][critical][backtrace] [external/envoy/source/server/backtrace.h:104] Caught Aborted, suspect faulting address 0x7
[2022-02-25 07:49:42.382][62][critical][backtrace] [external/envoy/source/server/backtrace.h:91] Backtrace (use tools/stack_decode.py to get line numbers):
[2022-02-25 07:49:42.382][62][critical][backtrace] [external/envoy/source/server/backtrace.h:92] Envoy version: 78625292510e31f0affd5311a2957c5ab2abd55c/1.20.1/Distribution>
[2022-02-25 07:49:42.382][89][critical][backtrace] [external/envoy/source/server/backtrace.h:98] #0: [0x7ffff7d10320]
[2022-02-25 07:49:42.382][89][critical][backtrace] [external/envoy/source/server/backtrace.h:98] #1: [0x5555585aff63]
[2022-02-25 07:49:42.382][62][critical][backtrace] [external/envoy/source/server/backtrace.h:98] #0: [0x7ffff7d10320]
[2022-02-25 07:49:42.382][89][critical][backtrace] [external/envoy/source/server/backtrace.h:98] #2: [0x7ffff7d5b49a]
[2022-02-25 07:49:42.382][89][critical][assert] [external/envoy/source/common/signal/signal_action.cc:43] panic: not implemented
 ```

### Steps to reproduce the bug

1. Setup on-prem k8s cluster v1.21.5
2. Apply 1.10.10 version of Gloo Edge as daemonset with hostNetwork that listen on ports 80,443


### Expected Behavior

That works the same as with version 1.10.8

### Additional Context

_No response_

**Comment by dakamarko** [comment]
The same issue with 1.10.9 as well

**Comment by kdorosh** [comment]
Following steps work fine for me:
- `kind version`: kind v0.11.1 go1.16.4 darwin/amd64
- `kind create cluster --name kind --image kindest/node:v1.21.1@sha256:69860bda5563ac81e3c0057d654b5253219618a22ec3a346306239bba8cfa1a6` (no newer kindest/node version to match 1.21.5, recommended version taken from [here](https://github.com/kubernetes-sigs/kind/releases/tag/v0.11.1))
- `glooctl install gateway --version 1.10.10`
- follow hello world steps as documented [here](https://docs.solo.io/gloo-edge/latest/guides/traffic_management/hello_world/), all works as expected. No envoy restarts even after making successful requests

Can you reproduce on `kind`? I don't have info (or perhaps ability) to reproduce your on-prem k8s cluster. Can you share the exact helm values you used for install? What steps did you take to produce the error after install?

**Comment by jack0** [comment]
I can reproduce this error on our own on-prem environment using gloo 1.10.[9,10,11,12] but 1.10.8 is fine. I am using the exact values.yaml for each install that I unfortunately cannot share.

**Comment by gunnar-solo** [comment]
I'm struggling to reproduce, similar to @kdorosh.

```

**Comment by pandarrr** [comment]
I'm seeing a similar issue. I could reproduce with just the docker compose example. I see the issue with EE version 1.10.7+ and OSS version 1.10.9+.

**Comment by gunnar-solo** [comment]
resolved with releases 
**gloo OSS**:
`v1.10.17`, `v1.9.10`, `v1.8.26`

**gloo EE**:
`v1.10.15`, `v1.9.12`

Have elected to forgo the 1.8.26 EE patch, as 1.8 is to be unsupported soon.  Will re-open if anyone has issues during the remaining support window

**Comment by nrjpoddar** [comment]
@gunnar-solo what was the underlying issue here?

**Comment by gunnar-solo** [comment]
ah--fair.  Sorry for that lack of context

I found that `gloo` OSS has been failing to deploy using `docker compose` since release 1.10.9.  These failures started with the introduction of  https://github.com/solo-io/envoy-gloo/commit/d3e4e3df7b6e024d8963866ef882911f528b667f, which was (for all intents and purposes) a no-op.  Poking around/testing found that [frolvlad/alpine-glibc](https://hub.docker.com/r/frolvlad/alpine-glibc/) updated their `:latest` tag _just_ around the same time frame as we built our failing release.  This PR pins the docker version to what it was _before_ the failures began (and solves said failures).

A couple of caveats:
* the fix I _just_ intruduced generates a gradual drift between what we're using and whatever `frolvlad/alpine-glibc` chooses is their `:latest`.  Not a huge loss, I think, for the upside of consistent CI
* envoy itself [no longer uses said alpine image during their build process](https://github.com/envoyproxy/envoy/commit/bdd22a65b5f20d79704f27e30bf39302440118bc)
* the failure this is fixing was not "everywhere".  It only occurred in specific deployment types.  For example, it passed our CI (kind), but failed docker-compose

This is "done', insofar as the crashing issue is fixed.  We have a new tech-debt issue of https://github.com/solo-io/gloo/issues/6084 to resolve the envoy drift
